lib/yaml_bot/rules_bot.rb

```diff
@@ -13,100 +13,91 @@ def initialize
     end
 
     def validate_rules
-      check_rules_are_loaded
+      fail_if_rules_are_not_loaded
       validate_root_keys
-      unless @missing_required_keys
-        @rules[:root_keys][:required].each do |key|
-          name = key.keys.first
-          unless key[name][:accepted_types].nil?
-            validate_accepted_types(key[name])
-          end
+      check_top_level_keys
+      YamlBot::Logging.info 'Rules file validated.'
+    end
 
-          unless key[name][:subkeys].nil?
-            validate_keys(key[name][:subkeys])
-          end
-        end
-      end
+    private
 
-      unless @missing_optional_keys
-        @rules[:root_keys][:optional].each do |key|
+    def check_top_level_keys
+      [:required, :optional].each do |key_type|
+        next if determine_key_type(key_type)
+        @rules[:root_keys][key_type].each do |key|
           name = key.keys.first
-          unless key[name][:accepted_types].nil?
-            validate_accepted_types(key[name])
-          end
-
-          unless key[name][:subkeys].nil?
-            validate_keys(key[name][:subkeys])
-          end
+          check_subkeys_or_accepted_types(key[name])
         end
       end
+    end
 
-      YamlBot::Logging.info 'Rules file validated.'
+    def check_subkeys_or_accepted_types(key)
+      validate_accepted_types(key) unless key[:accepted_types].nil?
+      validate_keys(key[:subkeys]) unless key[:subkeys].nil?
+    end
+
+    def determine_key_type(key_type)
```

Comment:
Need to change the method name to something more descriptive of what it's doing.